include/singa/io/kvfile.h

```diff
@@ -0,0 +1,146 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
we can split the code for kvfile and put the writing part into BinaryFileWriter, and put the reading part into BinaryFileReader (KVReader). Since all readers and writers process kv pairs, we can remove the prefix for sub-classes. When writing one record, we can use a 4 byte magic word as the boundary of a record.